### PR TITLE
Fix problem with remote job suite

### LINKED ISSUE #2369: problem with remote job suites (recent commit)
```
[scheduling]
   [[dependencies]]
       graph = foo
[runtime]
   [[foo]]
       [[[remote]]]
            host = localhost
            owner = hobo
```

Daemon aborts with;

```
ERROR - Traceback (most recent call last):
	  File "/home/vagrant/cylc.git/lib/cylc/task_job_mgr.py", line 813, in _prep_submit_task_job
	    job_conf = self._prep_submit_task_job_impl(suite, itask)
	  File "/home/vagrant/cylc.git/lib/cylc/task_job_mgr.py", line 915, in _prep_submit_task_job_impl
	    self.init_host(suite, itask.task_host, itask.task_owner)
	  File "/home/vagrant/cylc.git/lib/cylc/task_job_mgr.py", line 213, in init_host
	    proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
	  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
	    errread, errwrite)
	  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
	    raise child_exception
	OSError: [Errno 13] Permission denied

```

### PR DESCRIPTION
**Pull request by dvalters**
To close #2369

### CODE REVIEW

**Comment by dvalters** [comment]
@matthewrmshin could you check this with a few site-specific tests as well (for my sanity)

File: lib/cylc/task_job_mgr.py [hunk 1]
```diff
@@ -189,17 +189,19 @@ def init_host(self, reg, host, owner):
...
                 '-p',
                 self.suite_srv_files_mgr.get_contact_file(reg),
                 self.suite_srv_files_mgr.get_auth_item(
-                    self.suite_srv_files_mgr.FILE_BASE_PASSPHRASE, reg),
+                    self.suite_srv_files_mgr.FILE_BASE_PASSPHRASE,
+                    reg, owner, host),
                 user_at_host + ':' + r_suite_srv_dir + '/'])
             if ssl_cert is not None:
-                cmds.insert(-2, ssl_cert)
+                cmds.insert(-1, ssl_cert)
```
Comment (matthewrmshin) [inline lib/cylc/task_job_mgr.py]: Should be `cmds[-1].insert(-1, ssl_cert)`.

Comment (dvalters) [inline lib/cylc/task_job_mgr.py]: Aha, yes! - looking better now

**Comment by matthewrmshin** [comment]
I am totally wrong here. I don't think we need `owner` and `host` as arguments here - we are trying to get the local ones!

**Pull request merged**